docs/Developer Documentation/Logger/index.md

```diff
@@ -0,0 +1,159 @@
+
+
+Instance implementation : 
+```cpp
+static Logger& instance(Logger* pre_init = nullptr);
+
+Logger& Logger::instance(Logger* pre_init) {
+  static Logger* instance_ = nullptr;
+  if (pre_init) {
+    assert(instance_ == nullptr);
+    instance_ = pre_init;
+  }
+  assert(instance_);
+  return *instance_;
+}
+```
+
+`pre_init` is `nullptr` by default, so all components will access instance_ static pointer for logging. 
+`main()`  function neet to create SDLLogger implementation and call Logger::instance(logger_implementation);
+
+Plugin implementation:
+```cpp 
+extern "C" PluginType* Create(Logger* logger_singleton_instance) {
+  Logger::instance(logger_instance);
+  return new PluginType();
+}
+```
+
+SDL Core part will give pointer to logger singleton to the plugin so that plugin shared lib could initialize `Logger::instance` with the same pointer as core part. 
+
+## Logger detailed design :
+
+Each component creates `logger_` variable via macro. 
+This variable is actually a string with component name of the logger.
+Some logger implementations (like log4cxx) may have separate suverity or destanation  rules for each component. 
+
+Loggin macroses used to create such variable in each source file where loggin is required.
+
+SDL implements all info required for log message :
+
+ * LogLevel enum
+ * Location info struct : location in the code
+ * TimePoint 
+
+
+|||
+Detailed Design
+![Logger in details](./assets/detailed_logger_design.png)
+|||
+
+
+### Logger interface 
+
+Logger macroses use Logger interface for sending messages to External Logger. 
+Logger interface contains only methods required by any SDL component to perform logging : 
+
+ * PushLog(LogMessage)
+ * IsEnabledFor(LogLevel)
+ * Enabled()
+ * instance() - singleton
+
+
+### LoggerInitializer interface 
+
+LoggerInitializer specify interface required for Main to init/deinit logger but not required for any other SDL components
+LoggerInitializer should be templated with paticular ExternalLogger implementation
+
+It contains : 
+ - Init(ExternalLogger* external_logger)
+ - DeInit()
+
+Init and DeInit should perform internal initialization/deinitializetion and be proxied to ExternalLogger
+
+### ThirdPartyLogger interface
+
+ThirdPartyLogger interface extends Logger Interface with Initialisation and deinitialisation methods
+
+This interface should be inherited by paticulat external logger implementation 
+
+
+## Another logger implementation. 
+
+
+To use another (not log4cxx) logger, you should implement ThirdPartyLoggerInterface class 
+
+```cpp
+AnotherOneLoggerImpl : ThirdPartyLoggerInterface {
+  void SomeCustomMethod(parameters);
+  void Init() override;
+  void DeInit() override;
+  void Enable() override;
+  bool Enabled() override;
+  void Disable() override;
+  void Flush() override;
+  void PushLog(const LogMessage& log_message) override;
```

Comment:
Should be updated as some methods were renamed/removed

Reply:
Fixed

Reply:
fixed